### ISSUE TITLE
Import of shadowed module 'FMDB'

### ISSUE THREAD

**Issue by amantoux** [tekartik/sqflite#898]
When attempting to build on iOS, I get the following error message 
`Import of shadowed module 'FMDB'`

In my project, I'm using a Swift package that also depends on FMDB.

This is due to new file `SqfliteImport.h`
```objective-c
#ifndef SqfliteImport_h
#define SqfliteImport_h

#if TARGET_OS_IPHONE
#import <Flutter/Flutter.h>
#else
#import <FlutterMacOS/FlutterMacOS.h>
#endif

#import <fmdb/FMDB.h> // <- shadowed module due to this

#endif // SqfliteImport_h
```

I tried conditionally including FMDB header unsuccessfully
```objective-c
#ifndef FMDBReturnAutoreleased
#import <fmdb/FMDB.h>
#else
@import FMDB;
#endif
```

Any idea how to work around this?

**Comment by alextekartik** [comment]
Thanks for report. Unfortunately I'm still very bad at iOS stuff and I walk on eggs when I write native code here and I'm sometimes at lost at what to do. If you have a project to share so I can reproduce I would try to see if there is any change that could make sense here.

**Comment by alextekartik** [comment]
Also can you try to move the `#import <fmdb/FMDB.h>` out of this import file (and put it only where needed). Also in Xcode, can you try to remove the target membership of this file (SqliteImport.h)? (sorry but since I cannot reproduce I'm trying ultra remote debugging !)

**Comment by amantoux** [comment]
> Thanks for report. Unfortunately I'm still very bad at iOS stuff and I walk on eggs when I write native code here and I'm sometimes at lost at what to do. If you have a project to share so I can reproduce I would try to see if there is any change that could make sense here.

I've created an repo with and example that reproduces the problem
https://github.com/amantoux/test_sqflite.git

It simply added FMDB in the package dependency of the Xcode project of a new Flutter project.

<img width="1742" alt="Capture d’écran 2022-11-06 à 12 38 22" src="https://user-images.githubusercontent.com/17318211/200168415-567cd44c-b36d-4731-a187-292c625c2734.png">

**Comment by amantoux** [comment]
> Also can you try to move the `#import <fmdb/FMDB.h>` out of this import file (and put it only where needed). Also in Xcode, can you try to remove the target membership of this file (SqliteImport.h)? (sorry but since I cannot reproduce I'm trying ultra remote debugging !)

I tried moving the import into other files but it failed

**Comment by amantoux** [comment]
After playing with the example repo, it seem importing the `FMDB` **module** where necessary works.

Simply adding the following in `SqfliteCursor.h` and `SqflitePlugin.h`
```objective-c
@import FMDB;
```

**Comment by alextekartik** [comment]
Thanks for taking the time to write a program example and for proposing a solution. Sorry I made the change at the same time as your PR (which is quite simpler) but since I did it for ios and macos that's kind of too late for your PR.

I checked on my side that it works for projects that include or do not include FMDB. I published v2.2.1-1 (i.e. for now you have to explicitely set the following dependencies 'sqflite: '>=2.2.1-1') so that I can check on other projects that everything is fine.

**Comment by alextekartik** [comment]
I tested on some applications I have and unfortunately, I get the error `Module 'FMDB' not found` so unfortunately that does not look like a reasonable solution as most applications will get this error (and believe me, people yell when something break). I wonder whether we could make this compatible in both scenario (i.e. with and without a project importing FMDB), your input is welcome !

**Comment by amantoux** [comment]
Oh I believe you about the yelling!

Can you point me to the applications that break?
Since you have declared a dependency to FMDB in the podspec I would've assumed the module import would be enough (but I'm far from being an expert)

**Comment by alextekartik** [comment]
I tried on several applications and it works and of course it fails on a old one that I cannot share unfortunately and I cannot find a way to work around "Module FMDB not found" (and the Podfile.lock looks ok, I tried deleting Podfile file and Pods, .symlink directory, clean from flutter and xcode without much success). That's typically a case - old flutter project - where I don't want anybody to be stuck in. Well at least I have a reproducible case. I'll investigate later

**Comment by alextekartik** [comment]
(Sorry I share my lame progress). Ok I found a solution in my old projects. I needed to add `use_frameworks!`. Somehow this is properly added in Podile in new project but not in old one (even after deleting the Podfile).

I added a trouble shooting section on that here:https://github.com/tekartik/sqflite/blob/master/sqflite/doc/troubleshooting.md#module-fmdb-not-found

I'll test again on old projects I have in the next few days before releasing to stable (because I know I'll get issues about that). In the meantime you'll have to explicitly depends on `sqflite: '>=2.2.1-1'`.

**Comment by amantoux** [comment]
Many thanks!

**Comment by alextekartik** [comment]
Hi @amantoux Can you try with `sqflite: '>=2.2.2-1'`. @xtuck proposed a new solution as using `@import` was not working for him and it seems that his solution solves this for everyone (see #908)

**Comment by amantoux** [comment]
Hi @alextekartik,
Unfortunately, it doesn't work for me. I still have the shadowed module issue with `sqflite: '>=2.2.2-1'`

**Comment by xtuck** [comment]
@alextekartik  I think if need compatible both scene, "@import FMDB;" can't write in any header file (SqfliteFmdbImport.h).
when i write nether code in all .m file. it work successed.
#if __has_include(<fmdb/FMDB.h>)
#import <fmdb/FMDB.h>
#else
@import FMDB;
#endif

**Comment by xtuck** [comment]
@alextekartik sqflite: 2.2.2-1 can work for non-swift project(my app), 
but can't work for your test project: https://github.com/alextekartik/sqflite_issue_898_ios,
when i move the import code in SqfliteFmdbImport.h to all .m file, your project can work.

**Comment by alextekartik** [comment]
Ok thanks. Stupid me. Sorry @xtuck for trying a bad optimization and thanks @amantoux for testing too. I have published 2.2.2-2

**Comment by amantoux** [comment]
I confirm my project builds with 2.2.2-2.

**Comment by alextekartik** [comment]
Thanks @amantoux ! good to have ios developer to validate stuff. I tried a third attempt with release `2.2.2-3` since i hate duplicating code (especially when I don't fully understand it). Basically @xtuck I rename the .h file as .m file that I import from all other .m files, not such that it is a common practice but it sounds better to me rather than having 4 times the same import code and easier if we have to change in the future.

I also added a macOS/ios github actions to test that a project importing sqflite build using swift and objective C only (legacy): https://github.com/tekartik/sqflite/actions/workflows/run_ci_macos.yml to ensure I won't break this again in the future.

If all is well, I'll likely publish as stable today or tomorrow
